2023-risks/m1-insecure-authentication-authorization

```diff
@@ -0,0 +1,124 @@
+---
+
+layout: col-sidebar
+title: "M1: Insecure Authentication/Authorization"
+---
+
+# Threat Agents
+
+**Application Specific**
+
+Threat agents that exploit authentication and authorization vulnerabilities typically do so through automated attacks that use available or custom-built tools.
+
+# Attack Vectors
+
+**Exploitability EASY**
+
+Once the adversary understands the vulnerabilities in either the authentication or authorization scheme, they can exploit these weaknesses in one of two ways. They may either fake or bypass the authentication by directly submitting service requests to the mobile app's backend server, circumventing any direct interaction with the mobile app, or they can log into the application as a legitimate user after successfully passing the authentication control and then force-browse to a vulnerable endpoint to execute administrative functionality. Both exploitation methods are typically accomplished via mobile malware within the device or botnets owned by the attacker.
+
+# Security Weakness
+
+**Prevalence COMMON** <br />
+**Detectability AVERAGE**
+
+In order to test for poor authorization and authentication schemes in mobile apps, a number of strategies can be employed by testers. For authorization, testers can perform binary attacks against the mobile app and try to execute  privileged functionality that should only be executable with a user of higher privilege, particularly while the mobile app is in 'offline' mode. Testers should also attempt to execute any privileged functionality using a low-privilege session token within the corresponding POST/GET requests for the sensitive functionality to the backend server.
```

Comment:
I would suggest to test this vulnerability without the indirection through a binary attack but using a proxy or even forging own requests, e.g., with postman.

Reply:
Wouldn't you need the binary to list trigger the requests in the first place?

Reply:
You would need to have the binary installed to observe its communication with a proxy, right. But that would be sufficient to learn the end points and paths, which you could then attack with more specialized tools than a manipulated app. At least that would be my approach. But I have no strong opinion on this one, so going directly for the binary would also be fine for me.